Python/initconfig.c

```diff
@@ -698,6 +701,7 @@ _PyConfig_InitCompatConfig(PyConfig *config)
     config->parse_argv = 0;
     config->site_import = -1;
     config->bytes_warning = -1;
+    config->warn_default_encoding = -1;
```

Comment:
I suggest to set it to 0 here, and set to 1 to if -X option or env var is used.

For new variables, the magic -1 value is not needed. I used it for PyConfig members which have an associated global configuration variable.

(And so there is no need to set it again in config_init_defaults().)